### ISSUE TITLE
automatically gather more information on block detection

### ISSUE THREAD

**Issue by darkk** [TheTorProject/ooni-probe#647]
Simple `mtr --report --tcp --port 443 blocked.ws` and `mtr --report --udp --port 53 8.8.8.8` in addition to plain `mtr --report --icmp example.net` may add valuable information to web_connectivity report when some network block happens.

`ooniprobe` may gather alike data as soon as it detects major difference between measurement & control.

**Comment by bassosimone** [comment]
Yes, very good suggestion! On a related note (perhaps of the note to self type) the traceroute engine in measurement-kit is very basic, and I'd like to be able to do some `mtr --report` ish from measurement-kit (I doubt this could work on mobile, at least to the full extent that `mtr` works on desktops).

**Comment by darkk** [comment]
BTW, `mtr` may be misleading too. I've observed `mtr` showing stars for the final hop when the final hop was just replying slooowly (say 7-20seconds) while studying the data on TCP PEP (Performance-enhancing proxy) of a 3G ISP.
Always-on tcpdump revealed that `mtr` behavior and saved me from producing a false claim :)

**Comment by darkk** [comment]
Some notes while observing DNS hijacking at AS41843:
1. There is DNS answer race (like TCP RST race to terminate connection in other networks) when "bad" domain is queried — so patient client gets two answers
2. Both A and AAAA queries point to well-known blockpage webserver
3. DNS requests to whole internet (say, RA.N.DO.M) are intercepted and spoofed response is returned for bad domsina
4. TCP/DNS requests to the internet (say, 8.8.8.8, 77.88.8.8, 208.67.222.222) are not intercepted
5. Latency — min(spoofed) 2ms, min(8.8.8.8) 36ms, min(77.88.8.8) 35ms, min(208.67.222.222) 75ms
6. curl -H 'Host: dns.google.com' https://google.com/resolve is not intercepted (obviously) either cloudfronted or not

So (2) and (3) are already gathered, gathering data for (1), (4) and (5) may be useful in some cases.

**Comment by darkk** [comment]
Some notes while observing RST injection at AS8997:
1. `curl https://rutracker.org/forum/index.php` fails due to RST injection
2. `curl https://RuTracker.org/forum/index.php` works due to case-sensitive SNI field comparison
3. `openssl s_client -connect rutracker.org:443` works as openssl does not send SNI by default
4. fun fact: `rutracker.org` is closer to me (in terms of TTL) than the [injector service](https://gist.github.com/darkk/ba1948f0405e133c4b0065d1c1a3ffee#file-http-rutracker-org-txt)
5. [splitting ClientHello](https://gist.github.com/darkk/89b982151067d4555575b394753d2420) or `GET` in the middle of domain name does not trigger RST, moreover, splitting `ClientHello` anywhere within the packet passes through the filter
6. splitting `Host: rutracker.org\r\n` header **before** `\r\n` still triggers redirect injection, although it's technically incorrect as `Host: rutracker.org` may become `Host: rutracker.org.example.org\r\n` in further packets

**Comment by darkk** [comment]
RIPE Atlas DNS hackathon have brought another awesome masterpiece: [Recursive DNS Server Fingerprint](https://recdnsfp.github.io/).

**Comment by darkk** [comment]
PTRs of infrastructural IPs are also useful as services may migrate and it may be good to have that information stored (that can also be done in the pipeline).

**Comment by darkk** [comment]
Some notes while observing DNS zone hijacking at some of recursors in [AS22047](https://stat.ripe.net/AS22047):
1. queries to riseup.net domain return NXDOMAIN response **without** any additional data
2. query to **really** non-existent domain under `net` TLD returns SOA record for the TLD

<details>

```
; <<>> DiG 9.9.5-9+deb8u10-Debian <<>> -t soa riseup.net @190.160.0.14
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 14098
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4000
;; QUESTION SECTION:
;riseup.net.                    IN      SOA

;; Query time: 11 msec
;; SERVER: 190.160.0.14#53(190.160.0.14)


; <<>> DiG 9.9.5-9+deb8u10-Debian <<>> -t soa riseup-21059-24778-32011.net @190.160.0.14
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 50719
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4000
;; QUESTION SECTION:
;riseup-21059-24778-32011.net.  IN      SOA

;; AUTHORITY SECTION:
net.                    900     IN      SOA     a.gtld-servers.net. nstld.verisign-grs.com. 1499795223 1800 900 604800 86400

;; Query time: 17 msec
;; SERVER: 190.160.0.14#53(190.160.0.14)
```
</details>

**Comment by darkk** [comment]
measurement-kit/measurement-kit#1311 brings the idea of fingerprinting https blockpage server checking supported TLS versions and ciphersuites.
